independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientImpl.java

```diff
@@ -192,7 +192,7 @@ public Vertx get() {
             options.setShared(true);
         }
 
-        var httpClientBuilder = this.vertx.httpClientBuilder().with(options);
+        var httpClientBuilder = this.vertx.httpClientBuilder().with(options).with(options.getPoolOptions());
```

Comment:
Is it expected that the nested options are not applied by Vert.x? I would expect all the nested options to be applied when you do `with(options)`?

Especially since the nested structure is completely hidden from you.

Reply:
In other words, while I'm all for a quick fix, I wonder if it's not a bug in Vert.x itself?

Reply:
It's a totally legitimate quetions which I also asked @cescoffier in chat before applying this. 

For now let's get it in and we can figure later out if we need to update Vert.x itself